gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/TimingEventUtils.java

```diff
@@ -47,6 +47,14 @@ static Map<String, String> getFlowMetadata(Config flowConfig) {
     return metadata;
   }
 
+  /**
+   * Retrieves a flowExecutionId from flowMetadata map. Throws NPE if flowExecutionId is missing to prevent proceeding
```

Comment:
`Map.get` returns `null` when not found: it doesn't throw NPE.  that said, I strongly advise that whenever we do potentially return null to make the return type `Optional`, so the type checker can warn callers and hopefully **prevent** runtime NPEs